# Conhecendo a base de dados

## Importando os dados


Importando a lib pandas:

In [15]:
import pandas as pd

Importando os dados da url com a base de dados, guardando na variável de nome dados.
Utilizando o separadador de ; para que os dados fiquem em colunas, pois foram importados de um csv:

In [16]:
url = 'https://raw.githubusercontent.com/alura-cursos/pandas-conhecendo-a-biblioteca/main/base-de-dados/aluguel.csv'
dados = pd.read_csv(url, sep=';')
dados

Tipo                Bairro  Quartos  Vagas  Suites  \
0                    Quitinete            Copacabana        1      0       0   
1                         Casa       Jardim Botânico        2      0       1   
2      Conjunto Comercial/Sala       Barra da Tijuca        0      4       0   
3                  Apartamento                Centro        1      0       0   
4                  Apartamento          Higienópolis        1      0       0   
...                        ...                   ...      ...    ...     ...   
32955                Quitinete                Centro        0      0       0   
32956              Apartamento           Jacarepaguá        3      1       2   
32957              Apartamento  São Francisco Xavier        2      1       0   
32958              Apartamento                Leblon        2      0       0   
32959  Conjunto Comercial/Sala                Centro        0      0       0   

       Area   Valor  Condominio    IPTU  
0        40  1700.0       500.0    60.0  
1       100  7000.0         NaN     NaN  
2       150  5200.0      4020.0  1111.0  
3        15   800.0       390.0    20.0  
4        48   800.0       230.0     NaN  
...     ...     ...         ...     ...  
32955    27   800.0       350.0    25.0  
32956    78  1800.0       800.0    40.0  
32957    48  1400.0       509.0    37.0  
32958    70  3000.0       760.0     NaN  
32959   250  6500.0      4206.0  1109.0  

[32960 rows x 9 columns]

Função que retorna as cinco primeiras linhas (registros) da tabela:

In [17]:
dados.head()

Tipo           Bairro  Quartos  Vagas  Suites  Area  \
0                Quitinete       Copacabana        1      0       0    40   
1                     Casa  Jardim Botânico        2      0       1   100   
2  Conjunto Comercial/Sala  Barra da Tijuca        0      4       0   150   
3              Apartamento           Centro        1      0       0    15   
4              Apartamento     Higienópolis        1      0       0    48   

    Valor  Condominio    IPTU  
0  1700.0       500.0    60.0  
1  7000.0         NaN     NaN  
2  5200.0      4020.0  1111.0  
3   800.0       390.0    20.0  
4   800.0       230.0     NaN

Se colocar o valor 10, por exemplo, ele vai retornar as 10 primeiras linhas:

In [18]:
dados.head(10)

Tipo           Bairro  Quartos  Vagas  Suites  Area  \
0                Quitinete       Copacabana        1      0       0    40   
1                     Casa  Jardim Botânico        2      0       1   100   
2  Conjunto Comercial/Sala  Barra da Tijuca        0      4       0   150   
3              Apartamento           Centro        1      0       0    15   
4              Apartamento     Higienópolis        1      0       0    48   
5              Apartamento     Vista Alegre        3      1       0    70   
6              Apartamento         Cachambi        2      0       0    50   
7       Casa de Condomínio  Barra da Tijuca        5      4       5   750   
8       Casa de Condomínio            Ramos        2      2       0    65   
9  Conjunto Comercial/Sala           Centro        0      3       0   695   

     Valor  Condominio    IPTU  
0   1700.0       500.0    60.0  
1   7000.0         NaN     NaN  
2   5200.0      4020.0  1111.0  
3    800.0       390.0    20.0  
4    800.0       230.0     NaN  
5   1200.0         NaN     NaN  
6   1300.0       301.0    17.0  
7  22000.0         NaN     NaN  
8   1000.0         NaN     NaN  
9  35000.0     19193.0  3030.0

Método para trazer as cinco últimas linhas:

In [19]:
dados.tail()

Tipo                Bairro  Quartos  Vagas  Suites  \
32955                Quitinete                Centro        0      0       0   
32956              Apartamento           Jacarepaguá        3      1       2   
32957              Apartamento  São Francisco Xavier        2      1       0   
32958              Apartamento                Leblon        2      0       0   
32959  Conjunto Comercial/Sala                Centro        0      0       0   

       Area   Valor  Condominio    IPTU  
32955    27   800.0       350.0    25.0  
32956    78  1800.0       800.0    40.0  
32957    48  1400.0       509.0    37.0  
32958    70  3000.0       760.0     NaN  
32959   250  6500.0      4206.0  1109.0

Trazer o tipo de dado de uma variável:

In [20]:
type(dados)

pandas.core.frame.DataFrame

No caso acima, ele traz que o tipo de dado da variável Dados é Dataframe, pois retorna o valor: 
pandas.core.frame.DataFrame

## Características gerais da base de dados

O comando dados.shape retorna duas coordenadas.
A primeira é a quantidade de linhas do dataframe e a segunda é a quantidade de colunas.
No caso abaixo, teremos 32960 linhas e 9 colunas.

In [21]:
dados.shape

(32960, 9)

O comando dados.columns executado abaixo traz o nome das 9 colunas da nossa base de dados.

In [22]:
dados.columns

Index(['Tipo', 'Bairro', 'Quartos', 'Vagas', 'Suites', 'Area', 'Valor',
       'Condominio', 'IPTU'],
      dtype='object')

O comando dados.info() abaixo mostra as informações dos tipos de dados de cada coluna do dataframe:

In [23]:
dados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32960 entries, 0 to 32959
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Tipo        32960 non-null  object 
 1   Bairro      32960 non-null  object 
 2   Quartos     32960 non-null  int64  
 3   Vagas       32960 non-null  int64  
 4   Suites      32960 non-null  int64  
 5   Area        32960 non-null  int64  
 6   Valor       32943 non-null  float64
 7   Condominio  28867 non-null  float64
 8   IPTU        22723 non-null  float64
dtypes: float64(3), int64(4), object(2)
memory usage: 2.3+ MB


O comando abaixo dados['Tipo'] retornará uma series, uma estrutura dos Pandas que é mais simples que um dataframe, já que é composta apenas por índices e valores.
Em uma series vamos ter o índice na esquerda e depois os valores correspondentes aos índices na direita.

Perceba, então, que os dados da coluna "Tipo", são strings, ou seja, dados textuais.

In [24]:
dados['Tipo']

0                      Quitinete
1                           Casa
2        Conjunto Comercial/Sala
3                    Apartamento
4                    Apartamento
                  ...           
32955                  Quitinete
32956                Apartamento
32957                Apartamento
32958                Apartamento
32959    Conjunto Comercial/Sala
Name: Tipo, Length: 32960, dtype: object

O comando abaixo será dados[['Quartos', 'Valor']] , no qual ele vai retornar a coluna de nome Quartos e a coluna de nome Valor. 
Notamos, então, que os dados do tipo int são números inteiros, que é o caso da coluna "Quarto". Já os dados do float, da coluna "Valor", são decimais.

In [ ]:
dados[['Quartos', 'Valor']]

# Análise exploratória de dados

## Qual o valor médio de aluguel por tipo de imóvel?

## Removendo os imóveis comerciais

## Qual o percentual de cada tipo de imóvel na nossa base de dados?

### **Selecionando apenas os imóveis do tipo apartamento**

# Tratando e filtrando os dados

## Lidando com dados nulos

## Removendo registros

## Filtros

### **1. Apartamentos que possuem `1 quarto` e `aluguel menor que 1200`**



### **2. `Apartamentos` que possuem pelo menos `2 quartos`, `aluguel menor que 3000` e `area maior que 70`**

## Salvando os dados

# Manipulando os dados

## Criando colunas numéricas

## Criando colunas categóricas